### DISCUSSION TITLE
Add support for compression for remote logs

### DISCUSSION THREAD

**Discussion by malthe** [apache/airflow#27536] [General]
### Description

Since remote logs are sent only on completion, Airflow should support compressing them to save disk space.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
I will convert it into discussion. this is far from clear what expectation we have here.

What do you mean by remote logs in this context? Airflow has a number of handlers to send the remote logs to and how they are implemented depends on the system to send the logs to. Some habdlers handle only sending full logs (S3) and some are streaming (cloudwatch). Also I am not sure what do you mean by "compression" in this case. I'd understand if it was for actual local stored logs (but even then compression should be done by external scripts rather than by Airflow IMHO - or at the worst case if someone wnats they can install and configure logger handler such as this one https://pypi.org/project/compressedlogger/  - not sure why we would have to implement something like that in Airflow.

Could you please explain what do you mean by the request ? 

Converting into a discussion until this is clear :)

**Reply by malthe** [reply]
Let me elaborate a little. We're sending logs after a task completes to Azure Blob Storage.

They're just stored there in plain text. Unfortunately, none of the major object stores support something like "transparent compression" where text files can be stored cheaply so it's up to the client to compress files before sending.

What I'm asking for is that Airflow's object store remote log shipping handlers should grow a feature where they're able to:

A. Compress logs before sending
B. Understand when a log is compressed when serving them via the webserver

**Reply by potiuk** [reply]
But each remote store handler is implemented independently from each other (both sending and retrieval).  This is IMHO a feature of Azure Handler https://airflow.apache.org/docs/apache-airflow-providers-microsoft-azure/stable/_api/airflow/providers/microsoft/azure/log/wasb_task_handler/index.html#airflow.providers.microsoft.azure.log.wasb_task_handler.WasbTaskHandler not a feature of  Airflow (and needs no feature nor discussion, just implementation)

**Reply by potiuk** [reply]
Here also worth looking: https://airflow.apache.org/docs/apache-airflow-providers/core-extensions/logging.html there are other task handlers that also support streaming. the interface of those handlers is very well defined and how/where/what they do with the logs is up to implementation of that particular handler.
